src/shell/autoexec.cpp

```diff
@@ -507,9 +516,163 @@ void AutoExecModule::AddLine(const Placement placement, const std::string& line)
+// value types
+std::optional<AutoMountSettings> parse_drive_conf(const std_fs::path& conf_path)
+{
+	AutoMountSettings settings = {};
+
+	// If the conf path doesn't exist, at least return the default quiet arg
+	if (!path_exists(conf_path)) {
+		return settings;
+	}
+
+	// If we couldn't parse it, return the defaults
+	const auto conf = specify_drive_conf();
+	assert(conf);
+	if (!conf->ParseConfigFile("auto-mounted drive", conf_path.string())) {
+		return settings;
+	}
+
+	const auto section = static_cast<Section_prop*>(conf->GetSection("drive"));
+
+	const auto override_drive = section->Get_string("override_drive");
+	if (override_drive.length() == 1 && override_drive[0] >= 'a' &&
+	    override_drive[0] <= 'y') {
+		settings.override_drive = override_drive;
+	} else if (!override_drive.empty()) {
+		LOG_ERR("AUTOMOUNT: %s: setting 'override_drive = %s' is invalid",
+		        conf_path.c_str(),
+		        override_drive.c_str());
+		LOG_ERR("AUTOMOUNT: The override_drive setting can be left empty or a drive letter from 'a' to 'y'");
+	}
+
+	settings.readonly = section->Get_bool("readonly");
+	settings.label    = section->Get_string("label");
+	settings.type     = section->Get_string("type");
+	settings.path     = section->Get_string("path");
+	settings.verbose  = section->Get_bool("verbose");
+
+	return settings;
+}
+
+// Build a command to mount a drive from a directory, based on the
+// provided settings
+std::string build_auto_mount_dir_cmd(const std::string_view dir_letter,
+                                     const std_fs::path& drive_path,
+                                     const std::optional<AutoMountSettings>& settings)
+{
+	auto command = CmdMount;
+
+	if (!settings.has_value() || settings->override_drive.empty()) {
+		command += dir_letter;
+	} else {
+		command += settings->override_drive;
+	}
+
+	command += " " + Quote + simplify_path(drive_path).string() + Quote;
+
+	if (settings.has_value()) {
+		if (!settings->type.empty()) {
+			command += " -t " + settings->type;
+		}
+
+		if (!settings->label.empty()) {
+			command += " -label " + settings->label;
+		}
+
+		if (settings->readonly) {
+			command += " -ro";
+		}
+
+		if (!settings->verbose) {
+			command += ToNul;
+		}
+	}
+
+	return command;
+}
+
+// Get all files in the given directory that have the specified extensions.
+std::vector<std_fs::path> get_files_by_ext(
+        const std_fs::path& drive_path,
+        const std::initializer_list<std::string_view>& extensions)
+{
+	std::vector<std_fs::path> paths = {};
+
+	for (const auto& entry : std_fs::directory_iterator(drive_path)) {
+		if (entry.exists() && entry.is_regular_file()) {
+			const auto& path = entry.path();
+			std::string ext  = path.extension();
+			lowcase(ext);
+			if (std::ranges::find(extensions, ext) != extensions.end()) {
+				paths.push_back(path);
+			}
+		}
+	}
+
+	std::ranges::sort(paths, [](const std_fs::path& a, const std_fs::path& b) {
+		return a.filename() < b.filename();
+	});
```

Comment:
As mentioned above, use `natural_compare` here, just like the `IMGMOUNT` command.